api/controllers/UserController.js

```diff
@@ -120,5 +120,21 @@ module.exports = _.mapValues({
     const params = req.allParams();
     Validation.requireParams(params, 'name');
     return res.ok(await Validation.usernameAvailable(params.name));
-  }
+  },
+  async forgotUsername (req, res) {
+    const params = req.allParams();
+    Validation.requireParams(params, 'email');
+    const user = await User.findOne({email: params.email});
+    if (!user) return res.notFound();
```

Comment:
Remove this check; it would allow anyone to figure out whether a given email address is registered on Porybox.

Instead, we should send the response right after we validate that the request is valid, and then handle all the other stuff afterwards. That way we won't disclose whether the given email address actually exists.